### PR TITLE
Implement network task for AlibabaCloud

### PR DESCRIPTION
**Pull request by xh4n3**
Implement network task for AlibabaCloud
This is part of https://github.com/kubernetes/kops/issues/4127.

### CODE REVIEW

**Comment by xh4n3** [comment]
Travis CI breaks several times due to DNS issues, reopening again to run test.

File: upup/pkg/fi/cloudup/alitasks/vpc.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+}
+
+func (e *VPC) Run(c *fi.Context) error {
+	return fi.DefaultDeltaRunMethod(e, c)
+}
+
+func (_ *VPC) RenderALI(t *aliup.ALIAPITarget, a, e, changes *VPC) error {
+
+	if fi.BoolValue(e.Shared) && a == nil {
+		if a == nil {
```
Comment (andrewsykim) [inline upup/pkg/fi/cloudup/alitasks/vpc.go]: This extra if is not needed

File: upup/pkg/fi/cloudup/alitasks/vpc.go [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+	if a == nil {
+		if e.CIDR == nil {
+			return fi.RequiredField("CIDR")
+		}
+		if e.Name == nil {
+			return fi.RequiredField("Name")
+		}
+	}
+
+	if a != nil {
```
Comment (andrewsykim) [inline upup/pkg/fi/cloudup/alitasks/vpc.go]: Can just be `else` from the previous if

File: upup/pkg/fi/cloudup/aliup/ali_cloud.go [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+func (c *aliCloudImplementation) Region() string {
+	return c.region
+}
+
+func (c *aliCloudImplementation) ProviderID() kops.CloudProviderID {
+	return kops.CloudProviderALI
+}
+
+func (c *aliCloudImplementation) DNS() (dnsprovider.Interface, error) {
+	return nil, fmt.Errorf("DNS not implemented on aliCloud")
```
Comment (andrewsykim) [inline upup/pkg/fi/cloudup/aliup/ali_cloud.go]: nit: these can all be `errors.New()`

File: upup/pkg/fi/cloudup/aliup/ali_cloud.go [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+		VpcId:    id,
+	}
+	vpcs, _, err := c.EcsClient().DescribeVpcs(request)
+	if err != nil {
+		return nil, fmt.Errorf("error listing VPCs: %v", err)
+	}
+
+	if len(vpcs) != 1 {
+		return nil, fmt.Errorf("found multiple VPCs for %q", id)
+	} else {
```
Comment (andrewsykim) [inline upup/pkg/fi/cloudup/aliup/ali_cloud.go]: You can remove this `else`

File: upup/pkg/fi/cloudup/aliup/ali_cloud.go [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+
+const TagClusterName = "KubernetesCluster"
+
+// This is for statistic purpose.
+var KubernetesKopsIdentity = fmt.Sprintf("Kubernetes.Kops/%s", prj.Version)
+
+type ALICloud interface {
+	fi.Cloud
+
+	EcsClient() *ecs.Client
```
[the remark this answers is not in the document]
Comment (LilyFaFa) [reply upup/pkg/fi/cloudup/aliup/ali_cloud.go]: Thanks for your suggestions. I made corrections according to your suggestions. But about this one, we don't have such an interface for now, we'll try to add it in our SDK.

**Comment by xh4n3** [comment]
@andrewsykim Can we have a ok-to-test now?

**Comment by xh4n3** [comment]
@andrewsykim Thanks!
Is it okay to merge now?

**Pull request merged**